From d9a408bd58b4b2f480ea2ee010bb7559c9f2e93b Mon Sep 17 00:00:00 2001
From: Josh Wilsdon <josh@wilsdon.ca>
Date: Tue, 13 Feb 2018 18:46:27 -0800
Subject: [PATCH] OS-6687 want sysinfo key for bhyve support Reviewed by:
 Patrick Mooney <patrick.mooney@joyent.com> Reviewed by: Dave Eddy
 <dave.eddy@joyent.com> Approved by: Dave Eddy <dave.eddy@joyent.com>

---
 src/sysinfo | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/src/sysinfo b/src/sysinfo
index 653fcc62..c648e182 100644
--- a/src/sysinfo
+++ b/src/sysinfo
@@ -1,6 +1,6 @@
 #!/bin/bash
 #
-# Copyright (c) 2015 Joyent Inc., All rights reserved.
+# Copyright (c) 2018 Joyent Inc., All rights reserved.
 #
 # Output some info representing the system.  Default: JSON
 #
@@ -197,6 +197,16 @@ function get_smartos_cpu_info()
     #echo "${CPU_Cores}"
 }
 
+function get_bhyve_capability()
+{
+    Bhyve_Capable="false"
+    if [[ -x /usr/lib/brand/bhyve/bhhwcompat ]]; then
+        if /usr/lib/brand/bhyve/bhhwcompat; then
+            Bhyve_Capable="true"
+        fi
+    fi
+}
+
 function get_live_image_buildstamp()
 {
     # Add joyent buildstamp to SYSTEM_INFO
@@ -655,6 +665,7 @@ VM_Capable='${VM_Capable}'
 CPU_Type='${CPU_Version}'
 CPU_Virtualization='${CPU_Virtualization}'
 CPU_Physical_Cores=${CPU_Count}
+Bhyve_Capable='${Bhyve_Capable}'
 Nic_Tags=${NicTagList}
 Setup='${Setup_complete}'
 END
@@ -787,6 +798,7 @@ END
   "HW Family": "${HW_Family}",
   "Setup": "${Setup_complete}",
   "VM Capable": ${VM_Capable},
+  "Bhyve Capable": ${Bhyve_Capable},
   "CPU Type": "${CPU_Version}",
   "CPU Virtualization": "${CPU_Virtualization}",
   "CPU Physical Cores": ${CPU_Count},
@@ -975,6 +987,7 @@ else
 fi
 get_memory_mib
 get_smartos_cpu_info
+get_bhyve_capability
 get_live_image_buildstamp
 get_system_type
 get_hostname
-- 
2.21.0

